### PR TITLE
Testing with mkdir/create directory & try to use ObjectMapper

### PR DESCRIPTION
**Pull request by saleto**
I'm testing ObjectMapper, and try to create package like sample json structure.

### CODE REVIEW

File: src/main/java/pb360/json/JacksonObjectMapper.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import pb360.model.RestAPI;
+
+public class JacksonObjectMapper {
+	public static void main(String[] args) throws IOException{
+
+		try {
+		ObjectMapper objectMapper = new ObjectMapper();
+		RestAPI restApi =objectMapper.readValue(new File("D:\\restapi\\JSON_Sample.json"), RestAPI.class);
```
Comment (hoan-salt) [inline src/main/java/pb360/json/JacksonObjectMapper.java:19]: help me explain about the function of this class, maybe it is not using on the project

File: src/main/java/pb360/service/OptionsService.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package pb360.service;
+
+public interface OptionsService {
```
Comment (hoan-salt) [inline src/main/java/pb360/service/OptionsService.java]: why we need add this class at here

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
+	public String ControllerName;
+	public String link = "D:\\restapi\\JSON_Sample.json";
 
 	@Override
 	public MessageObject getRestApiData(String id) {
 		MessageObject getMessage = new MessageObject();
 		getMessage.setData("Get RestApi data successful: " + id);
 		getMessage.setType("GET");
 		getMessage.setDatetime(getCurrentDateTime().getDatetime());
+		ReadJson(link);
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java]: please flow the rule of java function name

File: src/main/java/pb360/validation/ValidateRestAPI.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			messageObj.setType("Missing Field");
+			messageObj.setData("restId");
+		}
+
+		if (restApi.getRestName() == null) {
+			messageObj.setType("Missing Field");
+			messageObj.setData("restName");
+		}
+
+		if (restApi.getRestStatus() == null) {
```
Comment (hoan-salt) [inline src/main/java/pb360/validation/ValidateRestAPI.java]: this is wrong

File: src/main/java/pb360/validation/ValidateRestAPI.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+			messageObj.setType("Missing Field");
+			messageObj.setData("restStatus");
+		}
+
+		if (restApi.getRestUrl() == null) {
+			messageObj.setType("Missing Field");
+			messageObj.setData("restUrl");
+		}
+
+		if (restApi.getRestLocation() == null) {
```
Comment (hoan-salt) [inline src/main/java/pb360/validation/ValidateRestAPI.java]: this is wrong, please remove

File: src/main/java/pb360/validation/ValidateRestAPI.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+			messageObj.setType("Missing Field");
+			messageObj.setData("restLocation");
+		}
+
+		if (restApi.getLastModifying() == null) {
```
Comment (hoan-salt) [inline src/main/java/pb360/validation/ValidateRestAPI.java]: please remove

File: src/main/java/pb360/validation/ValidateRestAPI.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+			messageObj.setType("Missing Field");
+			messageObj.setData("lastModifying");
+		}
+
+		if (restApi.getFileOfRest() == null) {
```
Comment (hoan-salt) [inline src/main/java/pb360/validation/ValidateRestAPI.java]: please remove

File: src/main/java/pb360/service/optionImpl/AbstractOptionsImpl.java [hunk 8]
```diff
@@ -1,5 +1,5 @@
 package pb360.service.optionImpl;
 
-public class Options {
+public class AbstractOptionsImpl {
```
Comment (hoan-salt) [inline src/main/java/pb360/service/optionImpl/AbstractOptionsImpl.java]: remove class

File: src/main/java/pb360/model/embedded/line.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package pb360.model.embedded;
+
+public class line {
```
Comment (hoan-salt) [inline src/main/java/pb360/model/embedded/line.java]: why we need to create a class for store only variable?

File: src/main/java/pb360/model/JsonDataModel.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
+package pb360.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+@JsonInclude(Include.NON_DEFAULT)
+public class JsonDataModel {
```
Comment (hoan-salt) [inline src/main/java/pb360/model/JsonDataModel.java]: I explained about JsonDataObject, why it still wrong

File: src/main/java/pb360/data/entity/RestApi.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import pb360.model.MessageObject;
+
+@Document(collection= "RestApi")
+
+public final class RestApi {
+	private @JsonProperty("restId") String restId;
```
Comment (hoan-salt) [inline src/main/java/pb360/data/entity/RestApi.java]: why u added JsonProperties on an entry of mongoDb?

File: src/main/java/pb360/json/JacksonObjectMapper.java (hunk 1 above)
Comment (saleto) [reply src/main/java/pb360/json/JacksonObjectMapper.java:19]: i try to use this to read json file

File: src/main/java/pb360/service/OptionsService.java (hunk 2 above)
Comment (saleto) [reply src/main/java/pb360/service/OptionsService.java]: My mistake, i added as following to json file, i will delete it

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java (hunk 3 above)
Comment (saleto) [reply src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java]: i fixed it

File: src/main/java/pb360/validation/ValidateRestAPI.java (hunk 7 above)
Comment (saleto) [reply src/main/java/pb360/validation/ValidateRestAPI.java]: i removed all of them

File: src/main/java/pb360/model/embedded/line.java (hunk 9 above)
Comment (saleto) [reply src/main/java/pb360/model/embedded/line.java]: My idea was wrong so i delete this too

File: src/main/java/pb360/data/entity/RestApi.java (hunk 11 above)
Comment (saleto) [reply src/main/java/pb360/data/entity/RestApi.java]: So, JsonProperties  should only be on Model right ?

File: src/main/java/pb360/model/JsonDataModel.java (hunk 10 above)
Comment (saleto) [reply src/main/java/pb360/model/JsonDataModel.java]: i will fix it

File: src/main/java/pb360/model/embedded/Annotation.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
+package pb360.model.embedded;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Annotation {// cung lam tuong tu luon, cai nao de lam truoc di e, da, thanks e, nen nho nhung khuc code doc file hoac xu ly co exception ta k ne n hardcode, nhung vay la app cua e chi chay cho e thoi, k ai xai dk, heiu k da em hieu roi, um thanks e!bye
```
Comment (hoan-salt) [inline src/main/java/pb360/model/embedded/Annotation.java]: remove commment in the code

File: src/main/java/pb360/data/entity/RestApi.java [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+	private @JsonProperty("restName") String restName;
+
+	private @JsonProperty("restUrl") String restUrl;
+
+	private @JsonProperty("restStatus") MessageObject restStatus;
+
```
Comment (hoan-salt) [inline src/main/java/pb360/data/entity/RestApi.java]: remove @JsonProperty() in the define variable scope, only add it on the constructor

File: src/main/java/pb360/model/RestAPI.java [hunk 14]
```diff
@@ -26,7 +23,7 @@
 
 	private @JsonProperty("lastModifying") Date lastModifying;
```
Comment (hoan-salt) [inline src/main/java/pb360/model/RestAPI.java]: please coding remove the @jsonproperty in the line define new variable! this is wrong

**Comment by hoan-salt** [comment]
Please remove bin files out PR

**Pull request merged**